controllers/storageclaim_controller.go

```diff
@@ -651,6 +673,70 @@ func (r *StorageClaimReconciler) hasVolumeSnapshotContents() (bool, error) {
 	return false, nil
 }
 
+func (r *StorageClaimReconciler) getCephDriverVolumeGroupSnapshotClass(driverType string,
+	data map[string]string) *groupsnapapi.VolumeGroupSnapshotClass {
+	var driverName string
+	if driverType == "cephfs" {
+		driverName = csi.GetCephFSDriverName()
+	} else {
+		driverName = csi.GetRBDDriverName()
+	}
+	volumegroupsnapshotclass := &groupsnapapi.VolumeGroupSnapshotClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: r.storageClaim.Name,
+		},
+		Driver:         driverName,
+		DeletionPolicy: snapapi.VolumeSnapshotContentDelete,
+		Parameters:     data,
+	}
+	return volumegroupsnapshotclass
+}
+
+func (r *StorageClaimReconciler) createOrReplaceVolumeGroupSnapshotClass(volumeGroupSnapshotClass *groupsnapapi.VolumeGroupSnapshotClass) error {
+	existing := &groupsnapapi.VolumeGroupSnapshotClass{}
+	existing.Name = r.storageClaim.Name
+
+	if err := r.own(volumeGroupSnapshotClass); err != nil {
+		return fmt.Errorf("failed to own volumegroupsnapshotclass: %v", err)
+	}
```

Comment:
Isn't volumeGroupSnapshotClass (and volumeSnapshotClass) a cluster-wide resource? In that case, you cannot own it by a namespace scoped resource (StorageClaim resource). We are already working around that using the storage claim annotation. So let's remove this `own` call.